File: app-geekhome/src/main/java/eu/geekhome/rest/hardware/HardwareAdapterDtoMapper.java
package eu.geekhome.rest.hardware;

import eu.geekhome.services.hardware.HardwareAdapter;
import eu.geekhome.services.hardware.HardwareAdapterDto;

import javax.inject.Inject;

public class HardwareAdapterDtoMapper {

    @Inject
    HardwareAdapterDtoMapper() {
    }

    public HardwareAdapterDto map(HardwareAdapter adapter) {
        return new HardwareAdapterDto(adapter.getState(), adapter.getLastDiscoveryTime(), adapter.getLastError().toString());
    }
}

File: app-geekhome/src/main/java/eu/geekhome/rest/hardware/HardwareAdapterDtoMapper.kt
package eu.geekhome.rest.hardware

import eu.geekhome.domain.hardware.AdapterBundle
import eu.geekhome.data.hardware.HardwareAdapterDto

class HardwareAdapterDtoMapper {

    fun map(bundle: AdapterBundle): HardwareAdapterDto {
        val lastError : String? = if (bundle.adapter.lastError != null) bundle.adapter.lastError.toString() else null

        return HardwareAdapterDto(
            bundle.adapter.id,
            bundle.owningPluginId,
            bundle.adapter.state,
            bundle.adapter.lastDiscoveryTime,
            lastError,
            bundle.adapter.ports.size)
    }
}